### ISSUE TITLE
No IncompatibleMethodOverride raised when base method is an overload

### ISSUE THREAD

**Issue by tmke8** [microsoft/pyright#5200]
**Describe the bug**
The following code has an incompatible override, but pyright raises no error.

**To Reproduce**
```python
from typing import Union, overload

class A:
    @overload
    def f(self, x: int) -> int: ...
    @overload
    def f(self, x: str) -> str: ...
    def f(self, x: Union[int, str]) -> Union[int, str]:
        return x

class B(A):
    def f(self) -> int:
        return 4

def g(a: A) -> str:
    return a.f("foo")

g(B())  # at runtime: TypeError: B.f() takes 1 positional argument but 2 were given
```
When I remove the overload, pyright raises the error as expected.

**Expected behavior**
pyright should catch invalid overrides with overloads.

**VS Code extension or command-line**
command-line

**Comment by erictraut** [comment]
This is by design, so I don't consider this a bug. It's a reasonable enhancement request.

**Comment by erictraut** [comment]
I've enhanced the `reportIncompatibleMethodOverride` diagnostic check to support the case where the overridden method is overloaded. This support will be in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.312, which I just published. It will also be included in the next insiders release of pylance.
